# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/arm64/CodeGenerator-arm64.cpp
# Commit: 09390cf1d667
# Full Hash: 09390cf1d66716146aeb3112f0ac640e317890d9
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-14 09:51:01
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.6: Add tls dependency for truncate i32. r=lth
#   
#   We generate builtin call for MTruncateToInt32 operation for floating points types,
#   so we need to add a tls dependency.
#   I inserted NYI for arm64 because Ion doesn't support arm64.
# ==============================================================================

diff -r 83a3ed07220d -r 09390cf1d667 js/src/jit/arm64/CodeGenerator-arm64.cpp
--- a/js/src/jit/arm64/CodeGenerator-arm64.cpp	Mon Sep 14 04:00:18 2020 +0000
+++ b/js/src/jit/arm64/CodeGenerator-arm64.cpp	Mon Sep 14 05:19:44 2020 +0000
@@ -1162,11 +1162,21 @@
                      ins->mir());
 }
 
+void CodeGenerator::visitWasmBuiltinTruncateDToInt32(
+    LWasmBuiltinTruncateDToInt32* lir) {
+  MOZ_CRASH("NYI");
+}
+
 void CodeGenerator::visitTruncateFToInt32(LTruncateFToInt32* ins) {
   emitTruncateFloat32(ToFloatRegister(ins->input()), ToRegister(ins->output()),
                       ins->mir());
 }
 
+void CodeGenerator::visitWasmBuiltinTruncateFToInt32(
+    LWasmBuiltinTruncateFToInt32* lir) {
+  MOZ_CRASH("NYI");
+}
+
 FrameSizeClass FrameSizeClass::FromDepth(uint32_t frameDepth) {
   return FrameSizeClass::None();
 }